interview-mr/src/main/java/com/sovrn/interview/mr/WebDomainScore.java

```diff
@@ -0,0 +1,102 @@
+package com.sovrn.interview.mr;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Iterator;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+
+public class WebDomainScore {
+	protected String category;
+	protected Float score;
+	protected String url;
+	protected String normUrl; // normalized
+	protected String company;
+
+	public static WebDomainScore fromTabSeparatedValues(String tsv) {
+		WebDomainScore newScore = new WebDomainScore();
+		
+		if (Strings.isNullOrEmpty(tsv)) {
+			return newScore;
+		}
+		
+		Iterator<String> tabs = Splitter.on('\t').split(tsv).iterator();
+		if (tabs.hasNext()) {
+			newScore.setCategory(tabs.next());
+		}
+		if (tabs.hasNext()) {
+			String scoreString = tabs.next();
+			try {
+				newScore.setScore(Float.parseFloat(scoreString));
+			} catch (NumberFormatException nfe) {
+				newScore.setScore(null);
+			}
+		}
+		if (tabs.hasNext()) {
+			newScore.setUrl(tabs.next());
+		}
+		if (tabs.hasNext()) {
+			newScore.setNormUrl(tabs.next());
+		}
+		if (tabs.hasNext()) {
+			newScore.setCompany(tabs.next());
+		}
+		return newScore;
+	}
+	
+	public String toTabSeparatedValues() {
+		Joiner tabJoiner = Joiner.on('\t').useForNull("");
+		return tabJoiner.join(getCategory(), getScore(), getUrl(), getNormUrl(), getCompany());
+	}
+
+	public String getHostname() {
+		if (Strings.isNullOrEmpty(normUrl)) {
+			return null;
+		}
+		
+		try {
+			URL wellFormedUrl = new URL(normUrl);
+			return wellFormedUrl.getHost();
+		} catch (MalformedURLException mfue) {
+			// TODO for production, handle malformed urls
```

Comment:
How should we handle MalformedURLs in production? Is this possible if the URL is already normalized?